src/main/java/com/esri/core/geometry/ogc/OGCGeometry.java

```diff
@@ -328,6 +325,10 @@ public boolean relate(OGCGeometry another, String matrix) {
 
 	// analysis
 	public double distance(OGCGeometry another) {
+		if (another.geometryType() == OGCConcreteGeometryCollection.TYPE) {
```

Comment:
Shouldn't the same logic apply to other methods, e.g. `contains`, `overlaps`, etc.?

Reply:
Yes most need to be changed eventually.